spec/schema.spec.js

```diff
@@ -0,0 +1,77 @@
+            }
+
+            if (fn.match("number.json")) {
+                return promiseMock.resolve(new Buffer('{"id": "number", "type": "number", "minimum": 5}', "binary"));
+            }
+
+            if (fn.match("email-no-id.json")) {
+                return promiseMock.resolve(new Buffer('{"type": "string", "format": "email"}', "binary"));
+            }
+
+            if (fn.match("email-parse-error.json")) {
+                return promiseMock.resolve(new Buffer('{"type: "string", "format": "email"}', "binary"));
+            }
+
+            return promiseMock.reject("Invalid file: " + fn.toString());
+        });
+        nodeValidator = require("validator");
+        promiseMock = require("./mock/promise-mock");
+        tv4 = require("tv4");
+        schema = require("../lib/schema")(fs, nodeValidator, promiseMock, tv4);
+    });
+    describe(".loadSchemaAsync()", () => {
+        beforeEach(() => {
+            tv4.addSchema = jasmine.createSpy("tv4.addSchema").andCallThrough();
+        });
+        it("loads a schema and validates against it", (done) => {
+            schema.loadSchemaAsync("./email.json").then(() => {
+                expect(schema.validate("someone@example.net", "email")).toBe(true);
+                expect(schema.validate("someone", "email")).toBe(false);
+            }).then(done, done);
+        });
+        it("loads a schema not containing an id", (done) => {
+            jasmine.testPromiseFailure(schema.loadSchemaAsync("./email-no-id.json"), "Schema did not contain id: ./email-no-id.json", done);
+        });
+        it("loads a schema which cannot be parsed", (done) => {
+            jasmine.testPromiseFailure(schema.loadSchemaAsync("./email-parse-error.json"), "Unble to parse file: ./email-parse-error.json", done);
+        });
+         it("tries to a schema which is not present", (done) => {
+            jasmine.testPromiseFailure(schema.loadSchemaAsync("./email-not-there.json"), "Unble to parse file: ./email-not-there.json", done);
+        });
+    });
+    describe(".loadSchemaFolderAsync()", () => {
+        it("loads schemas and validates against them", (done) => {
+            fs.readdirAsync.andCallFake((fn) => {
+                if (fn.match("/folder/")) {
+                    return promiseMock.resolve([
+                        "email.json",
+                        "number.json"
+                    ]);
+                }
+
+                return promiseMock.reject("Invalid folder: " + fn.toString());
+            });
+            schema.loadSchemaFolderAsync("/folder/").then(() => {
+                expect(schema.validate("someone@example.net", "email")).toBe(true);
+                expect(schema.validate("someone", "email")).toBe(false);
+                expect(schema.validate(5, "number")).toBe(true);
+                expect(schema.validate(2, "number")).toBe(false);
```

Comment:
These validate the validator more than the schema finder.  You don't need negative cases as much here, though what happens when you run `schema.validate({}, "whatever")`?